### PR TITLE
Add ability to save raw text of udf

### PR DESCRIPTION
**Pull request by Shelnutt2**
Add ability to save raw text of udf. This is disableable with the do_not_include_source_lines parameter.

### CODE REVIEW

File: tiledb/cloud/array.py [hunk 1]
```diff
@@ -254,6 +255,7 @@ def apply_async(
     layout=None,
     image_name=None,
     http_compressor="deflate",
+    do_not_include_source_lines=False,
```
Comment (ihnorton) [inline tiledb/cloud/array.py]: -> `include_source_lines=True` and avoid the double negative below?

File: tiledb/cloud/array.py [hunk 2]
```diff
@@ -284,10 +287,23 @@ def apply_async(
         raise TypeError("name argument to `apply` must be set if no function is passed")
 
     pickledUDF = None
+    source_lines = None
     if func is not None:
         pickledUDF = cloudpickle.dumps(func, protocol=udf.tiledb_cloud_protocol)
         pickledUDF = base64.b64encode(pickledUDF).decode("ascii")
 
+        if not do_not_include_source_lines:
```
Comment (ihnorton) [inline tiledb/cloud/array.py]: (change as suggested in :258?)

File: tiledb/cloud/udf.py [hunk 3]
```diff
@@ -232,6 +294,19 @@ def update_udf(func, name, namespace=None, image_name=None, type=None):
         pickledUDF = cloudpickle.dumps(func, protocol=tiledb_cloud_protocol)
         pickledUDF = base64.b64encode(pickledUDF).decode("ascii")
 
+        source_lines = None
+        if not do_not_include_source_lines:
```
Comment (ihnorton) [inline tiledb/cloud/udf.py]: 4th call I think -- suggest to make this a function?

**Comment by Shelnutt2** [comment]
@ihnorton I've made the suggested changes, please let me know if you like it better. The function was a good idea, was repeating the same lines in too many places.

File: tiledb/cloud/utils.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import inspect
+import logging
+
+logger = logging.getLogger("tiledb.cloud")
+
+
+def getsourcelines(func):
+    if func is not None and not callable(func):
```
Comment (ihnorton) [inline tiledb/cloud/utils.py]: oh, btw this can be written as `if not callable(func)` (`callable(None)` -> `False`)

**Pull request merged**